### PR TITLE
Update VecElement description

### PR DESCRIPTION
**Pull request by eschnett**
Remove restriction that does not apply any more.

### CODE REVIEW

File: doc/src/base/simd-types.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ end
 ```
 
 It has a special compilation rule: a homogeneous tuple of `VecElement{T}` maps to an LLVM `vector`
-type when `T` is a primitive bits type and the tuple length is in the set {2-6,8-10,16}.
```
Comment (KristofferC) [inline doc/src/base/simd-types.md]: `T` still has to be a primitive bits type thought, right?

Comment (ViralBShah) [reply doc/src/base/simd-types.md:13]: I would imagine T still has to be a primitive bits type.

**Comment by ViralBShah** [comment]
If this restriction on tuple length is no longer in place, it would be nice to update this PR accordingly and merge.

**Comment by eschnett** [comment]
Updated the PR.

**Pull request merged**